Exercise1/Exercise1/Services/MovieService.cs

```diff
@@ -0,0 +1,239 @@
+﻿using Exercise1.Models;
+using Exercise1.Repository;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Media;
+using System.Text;
+using System.Threading.Tasks;
+using Unity.Injection;
+
+namespace Exercise1.Services
+{
+    public class MovieService
+    {
+        private readonly IMovieRepository _movieRepository;
+        private readonly IStudioRepository _studioRepository;
+
+        public MovieService(IMovieRepository movieRepository, IStudioRepository studioRepository)
+        {
+            _movieRepository = movieRepository;
+            _studioRepository = studioRepository;
+        }
+        //add movie
+        public void AddNewMovie()
+        {
+            var movie = new Movie
+            {
+                Name = Validation("Enter the movie name: "),
+                Year = ValidationInt("Enter movie year: ", 1900, 2024),
+                Description = Validation("Enter movie descrption: "),
+                Genre = Validation("Enter movie genre: "),
+                Studios = new List<Studio>()
+            };
+            bool AddNewStudio = true;
+            while (AddNewStudio)
+            {
+                string studioName = Validation("enter the studio name; ");
+                if (!string.IsNullOrWhiteSpace(studioName))
+                {
+                    var studio = _studioRepository.GetStudio(studioName);
+
+                    if (studio == null)
+                    {
+                        Console.WriteLine("Studio not found. Do you want to add one y/n? ");
+                        if(Console.ReadLine()?.ToLower() == "y")
+                        {
+                            studio = new Studio
```

Comment:
the creation of a new object its not needed, you should be able to update the desired fields of the retrieved studio